sclang: AnalogOut and DigitalOut return 0.0

elgiano opened the pull request:
<!--- Hi, and thanks for contributing! -->
<!--- Make sure to provide a general summary of your changes in the title above. -->
<!--- For example: "[scsynth] Fix crash when encountering cute kittens" -->

Purpose and Motivation
----------------------
Fixes #61.
For UGens without any output, such as AnalogOut and DigitalOut, returning 0.0 prevents the creation of a wrapping Out UGen, which in turn was seen messing with poll rates in issue #61.
This technique was copied from other outputless UGens, such as SendTrig.

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to it here by writing "Fixes #555". -->

Types of changes
----------------

<!--- What types of changes does your pull request introduce? -->
<!--- Some examples are below (you can delete the lines that don't apply): -->

- Bug fix (non-breaking change which fixes an issue)

Checklist
---------

<!--- Complete an item by checking it: [x] -->

- [x] All tests are passing
- [x] This PR is ready for review

<!--- See DEVELOPING.md for instructions on running and writing tests. -->

Remaining Work
--------------

<!--- If any work remains to be done, please give a brief description here. -->
<!--- Consider providing a todo-list so we can easily track completion progress. -->

<!--- Thanks for contributing! -->

giuliomoro commented on SCClassLibrary/Common/Audio/bela/BELAUGens.sc:
```diff
@@ -68,10 +70,12 @@ DigitalIn : UGen {
     *ar { arg digitalPin = 0, output=0, writeMode=0, mul=1.0, add=0.0;
-        ^this.multiNew('audio', digitalPin, output, writeMode ).madd(mul,add)
+        this.multiNew('audio', digitalPin, output, writeMode ).madd(mul,add);
+		^0.0;
```
something weird with indentation. You should use `tools/clang-format.py` to lint your files prior to commit

elgiano replied:
Sorry, the problem is that clang-format doesn't format .sc files, and these files are all in spaces, while all .sc files in main SuperCollider are in tabs. I'll put those lines in spaces as well, then maybe we change all of them to tabs at once before merge with main sc?

giuliomoro replied:
yeah we made a mess the moment we started editing these ... what I do in these cases is I try to keep the convention of the surrounding code. I thought the linter would take care of these, but clearly it doesn't, thanks for letting me know and fixing it.

giuliomoro reviewed (commented):
No idea what this does :)  (because of my ignorance in Sc)

Fix the indentation and we should be fine! Is this needed also for the analog and digital ins ?

elgiano commented:
Analog and Digital In have outputs, so it should be all right as they are :)

elgiano committed:
sclang: AnalogOut and DigitalOut return 0.0

fixes an issue with Function::play. For UGens without any output, such 
as AnalogOut and DigitalOut, returning 0.0 prevents the creation of a 
wrapping Out UGen, which in turn was seen messing with poll rates in 
Bela's issue #61.

elgiano commented:
I see that DigitalIn/Out are not "supposed" to have modulatable pin index... shall I change this, make it modulatable and put a check there, or shall I leave it as it is?

giuliomoro commented:
I don't think that's needed... Or at least no one asked for it yet!

elgiano commented:
Ops, sorry I posted in the wrong PR, I was referring to your request in #76, I'll continue back there
